presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -587,8 +585,8 @@ public void checkCanShowColumns(SystemSecurityContext context, CatalogSchemaTabl
 
         Set<String> restrictedColumns = rule.getRestrictedColumns();
         return columns.stream()
-                .filter(columnMetadata -> !restrictedColumns.contains(columnMetadata.getName()))
-                .collect(toImmutableList());
+                .filter(column -> !restrictedColumns.contains(column))
+                .collect(toImmutableSet());
```

Comment:
This can be further simplified with `com.google.common.collect.Sets#difference`

(separate commit, if you decide to do so)